src/Ghcid.hs

```diff
@@ -162,15 +164,21 @@ autoOptions o@Options{..}
                                 map ("--ghci-options=" ++) opts
                             else
                                 "stack exec --test --bench -- ghci" : opts
-                in f flags $ stack:cabal
-              | ".ghci" `elem` files -> f ("ghci":opts) [curdir </> ".ghci"]
+                in updateOptions flags $ stack:cabal
+              | ".ghci" `elem` files -> updateOptions ("ghci":opts) [curdir </> ".ghci"]
               | cabal /= [] ->
                   let useCabal = ["cabal","repl"] ++ target ++ map ("--repl-options=" ++) opts
                       useGhci = "cabal exec -- ghci":opts
-                  in  f (if null arguments then useCabal else useGhci) cabal
-              | otherwise -> f ("ghci":opts) []
+                  in  updateOptions (if null arguments then useCabal else useGhci) cabal
+              | otherwise -> updateOptions ("ghci":opts) []
     where
-        f c r = o{command = unwords $ c ++ map escape arguments, arguments = [], restart = restart ++ r, run = [], test = run ++ test}
+        updateOptions command' restart' =
+          o { command = unwords $ command' ++ map escape arguments
+            , arguments = []
+            , restart = restart ++ restart'
+            , run = []
+            , test = run ++ test
+            }
```

Comment:
Drive-by fix to give these variables better names + format them a little more clearly